test/prob/test_fixture_cdf_log.hpp

```diff
@@ -354,11 +354,15 @@ class AgradCdfLogTestFixture : public ::testing::Test {
         << "Number of finite diff gradients and calculated gradients must "
            "match -- error in test fixture";
     for (size_t i = 0; i < finite_dif.size(); i++) {
-      EXPECT_NEAR(finite_dif[i], gradients[i], 1e-4)
-          << "Comparison of finite diff to calculated gradient failed for i="
-          << i << ": " << parameters << std::endl
-          << "  finite diffs: " << finite_dif << std::endl
-          << "  grads:        " << gradients;
+      // use relative error check in first case as logs of functions do not
```

Comment:
I'm familiar with how relative differences near zero fail and we switch to absolute errors, but I don't follow the log of functions thing here. Is it similar? Is this something special with gradients?

Also 10% seems like a really coarse error? I know things in this test framework are pretty rough but 0.1 seems really large.

Reply:
> Oh yeah is there a particular thing you need a CDF to be super robust for? Like is this for truncated distributions? Or for a link function? Might be worth throwing a comment in there saying something like these expansions look complicated but they're here because of X.

Yes the reason we came across this issue is because we were investigating the case of rare base-rate logistic regression problems in stan. We're proposing an alternative link function for dealing with these problems, which is able to more closely fit the analytic posteriors.

> @PhilClemson question -- @nhuurre pointed out this code might translate to other functions. This isn't something that needs done in this pull req, but do any of these expansions generalize? Clearly it's a lot of work to get them right and so it'd be nice if it could be boxed up at all.

Other than normal lccdf (which could be implemented as a simple y-axis reflection of normal lcdf) I'm not sure what else would be generalizable for use in other functions. The gradient approximations are definitely very specific and that's where most of the work has gone in.

Reply:
`ordered_probit_lpmf` uses `log(Phi(...))` directly so it can borrow this without any problem. 
It's more complicated for `skew_normal` and `exp_mod_normal`. Naive substitution where they use `log(erfc(...))` would result in catastrophic cancellation in the derivatives (in a subset of the region where they currently fail anyway...)

Reply:
@nhuurre are there problems where order_probit suffers from numeric breakdowns like happened here? If so I guess we make a separate issue out of this.